.github/workflows/push_dict_to_repos.yml

```
@@ -0,0 +1,95 @@
+name: Push dictionaries to OpenNeuro-JSONLD
+
+on:
+  push:
+    branches:
+      - main
+      - add_automation
+  workflow_dispatch:
+
+jobs:
+  update-datasets:
+    runs-on: ubuntu-latest
+    timeout-minutes: 720
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Find and submit data dictionaries
+        id: get_datasets
+        run: |
+
+          # Get the list of datasets
+          data_dictionaries=$(ls *.json)
+          n_dictionaries=$(echo "$data_dictionaries" | wc -l)
+
+          # Storage vars
+          success_count=0
+          fail_count=0
+          missing_repos=""
+          opened_prs=""
+
+          index=0
+          for dict in ${data_dictionaries}; do
+            index=$((index + 1))
+            dataset_name=$(basename $dict .json)
+            
+            # Attempt to upload the data dictionary with boilerplate information
+            response=$(curl --silent --retry 3 --retry-delay 30 \
+              -X 'PUT' \
+              "https://upload.neurobagel.org/openneuro/upload?dataset_id=${dataset_name}" \
+              -H 'accept: application/json' \
+              -H 'Content-Type: multipart/form-data' \
+              -F "data_dictionary=@${dict};type=application/json" \
+              -F 'changes_summary=Automatic upload from https://github.com/neurobagel/openneuro-annotations' \
+              -F 'name=Bot' \
+              -F 'email=bot@neurobagel.org' \
+              -F 'affiliation=neurobagel' \
+              -F 'gh_username=neurobagel-bot')
+
+            # Check the status of the response
+            if echo $response | jq -e 'has("error")' > /dev/null; then
+              error_message=$(echo "$response" | jq -r '.error')
+
+              if [[ "$error_message" =~ "same as in the target file." ]]; then
+                # Nothing to do
+                success_count=$((success_count + 1))
+                echo "(${index}/${n_dictionaries}) ${dataset_name}: ${error_message}"
+                continue
```

🍒 for handled case where JSON is invalid
```suggestion
              if [[ "$error_message" =~ "same as in the target file." ]]; then
                # Nothing to do
                success_count=$((success_count + 1))
                echo "(${index}/${n_dictionaries}) ${dataset_name}: ${error_message}"
                continue
              elif [[ "$error_message" =~ "not a valid JSON file" ]]; then
                fail_count=$((fail_count + 1))
                echo "(${index}/${n_dictionaries}) ${dataset_name}: ${error_message}"
                continue
```